lib/VideoInputTest.ts

```diff
@@ -0,0 +1,372 @@
+import { EventEmitter } from 'events';
+import {
+  AlreadyStoppedError,
+  DiagnosticError,
+  InvalidOptionsError,
+} from './errors';
+import {
+  getUserMedia,
+  GetUserMediaUnsupportedError,
+} from './polyfills';
+import {
+  SubsetRequired,
+  TimeMeasurement,
+  VideoResolution,
+} from './types';
+import {
+  InvalidityRecord,
+  validateDeviceId,
+  validateOptions,
+  validateTime,
+} from './utils/optionValidation';
+
+export declare interface VideoInputTest {
+  /**
+   * This event is emitted with a test report when the test ends.
+   * @param event [[VideoInputTest.Events.End]]
+   * @param report Summary of the test.
+   * @private
+   */
+  emit(
+    event: VideoInputTest.Events.End,
+    report: VideoInputTest.Report,
+  ): boolean;
+  /**
+   * This event is emitted with a [[DiagnosticError]] when the test encounters
+   * an error, fatal or not.
+   * @param event [[VideoInputTest.Events.Error]]
+   * @param error The [[DiagnosticError]] that was encountered.
+   * @private
+   */
+  emit(
+    event: VideoInputTest.Events.Error,
+    error: DiagnosticError,
+  ): boolean;
+
+  /**
+   * Raised upon completion of the test.
+   * @param event [[VideoInputTest.Events.End]]
+   * @param listener A callback that expects the following parameters:
+   *  An [[VideoInputTest.Report]] that summarizes the test.
+   * @returns This [[VideoInputTest]] instance.
+   * @event
+   */
+  on(
+    event: VideoInputTest.Events.End,
+    listener: (report: VideoInputTest.Report) => any,
+  ): this;
+  /**
+   * Raised by the test when encountering an error with a parameter of type
+   * [[DiagnosticError]].
+   * @param event [[VideoInputTest.Events.Error]]
+   * @param listener A callback that expects the following parameters:
+   *  A [[DiagnosticError]] that the test encountered.
+   * @returns This [[VideoInputTest]] instance.
+   * @event
+   */
+  on(
+    event: VideoInputTest.Events.Error,
+    listener: (error: DiagnosticError) => any,
+  ): this;
+}
+
+/**
+ * [[VideoInputTest]] class that parses options and starts a video input device
+ * test.
+ *
+ * Please see [[testVideoInputDevice]] for details and recommended practices.
+ */
+export class VideoInputTest extends EventEmitter {
```

Comment:
I wonder if there's enough shared between VideoInputTest and AudioInputTest to consider creating an InputTest base class and extending it?